utils/networking.js

```diff
@@ -17,57 +17,82 @@ const AD_REFRESH_INTERVAL = 15000;
 let prebidInit = false;
 let interval = null;
 const retryCount = 5;
-/** @type {HTMLIFrameElement} */
-let iframe = null;
-let ready = false;
-let bids = null;
-let heartbeatPending = false;
+let bids = {};
 const currentTries = {} // Maps retries to specific ad unit id
+const previousUrls = {} // Maps prior fetched URLs to specific ad unit id
+let baseDivId = 'pb-slot-right-1';
 
 const initPrebid = (adUnitId, format) => {
-  // Load zesty prebid iframe
-  iframe = document.createElement('iframe');
-  iframe.src = `https://www.zesty.xyz/prebid/?size=${format}&source=${Math.round(Math.random())}&ad_unit_id=${adUnitId}&utm_source=${adUnitId}`;
-  iframe.width = '1';
-  iframe.height = '1';
-  iframe.style.position = 'fixed';
-  iframe.style.border = 'none';
-  iframe.style.zIndex = '-2';
-  document.body.prepend(iframe);
-  iframe.onload = () => {
-    iframe.contentWindow.postMessage({ type: 'readycheck' }, '*');
-  }
-  window.addEventListener('message', ({ data }) => {
-    switch (data.type) {
-      case 'readystatus':
-        ready = data.content;
-        break;
-      case 'bids':
-        bids = data.content;
-        break;
-      case 'heartbeat':
-        heartbeatPending = false;
-        break;
-    }
-  });
+  // Create div for prebid to target
+  const div = document.createElement('div');
+  div.id = 'zesty-div';
+  div.style.height = '250px';
+  div.style.width = '300px';
+  div.style.position = 'fixed';
+  div.style.top = '0';
+  document.body.appendChild(div);
+
+  // Append google gpt tag
+  const script = document.createElement('link');
+  script.href = 'https://www.googletagservices.com/tag/js/gpt.js';
+  script.rel = 'preload';
+  script.as = 'script';
+  document.head.appendChild(script);
+
+  // Append aditude wrapper tag
+  const aditudeScript = document.createElement('script');
+  aditudeScript.src = 'https://dn0qt3r0xannq.cloudfront.net/zesty-ig89tpzq8N/zesty-longform/prebid-load.js';
+  aditudeScript.async = true;
+  document.head.appendChild(aditudeScript);
 
   // Load gifler script in case gif creative is served
-  const script = document.createElement('script');
-  script.src = 'https://cdn.jsdelivr.net/npm/gifler@0.1.0/gifler.min.js';
-  document.head.appendChild(script);
+  const gifscript = document.createElement('script');
+  gifscript.src = 'https://cdn.jsdelivr.net/npm/gifler@0.1.0/gifler.min.js';
+  document.head.appendChild(gifscript);
+
+  // Select baseDivId based on format, defaulting to the one for medium rectangle
+  if (format == 'billboard') {
```

Comment:
Could all of this information on each format's attributes be in `utils/formats.js`?

Reply:
It probably could yeah, can do in a small followup if this deploys without issue